### ISSUE TITLE
Request for appimaged to omit directory

### ISSUE THREAD

**Issue by tresf** [AppImage/AppImageKit#428]
A popular hypervisor "VirtualBox" uses `/opt/VBoxGuestAdditions-X.X.X` on Ubuntu Guests to store information about the VM guest additions.  At the time of writing this, this location has several files including 7 subfolders, source code and two desktop files.

Running `appimaged` in verbose mode shows [many failed attempts](https://gist.githubusercontent.com/tresf/1c080eaf7bb49c5682e9dee593669218/raw/348836c6831d29835a638cb4b92f649ba2e14ef7/appimaged.log) to scan and recognize files in this directory.

From what I understand this is due to the monitoring of the `/opt` directory.  I propose to omit/blacklist directories starting with `/opt/VBoxGuestAdditions-` to prevent common unwanted entries in startup.

Does such a blacklisted location already exist within  the tool?

**Comment by probonopd** [comment]
Relevant code is [here](https://github.com/AppImage/AppImageKit/blob/edb642d8879ec0a3ce8a67158e895f74310701f0/appimaged.c#L307-L316) so it would be easy to comment out `/opt` in your own builds, but no, there is no blacklist mechanism yet. PR would be appreciated though.

**Comment by TheAssassin** [comment]
[Blacklisting doesn't solve the problems of the world](https://www.schneier.com/blog/archives/2011/01/whitelisting_vs.html). I don't see the reason to monitor `/opt` anyway. How often is this directory used for AppImages? I think we can just get rid of it completely.

**Comment by tresf** [comment]
> There is no blacklist mechanism yet. 

It appears `inotifytools_watch_recursively` would have to be replaced with [`inotifytools_watch_recursively_with_exclude`](http://inotify-tools.sourceforge.net/api/inotifytools_8h.html#20981c7f1410d458b5ef0654b9adf1e4).

**Comment by probonopd** [comment]
Yes, `inotifytools_watch_recursively_with_exclude` looks like it @tresf.
`/opt` is a good option imho for system-wide AppImages, @TheAssassin.

**Comment by probonopd** [comment]
@tresf I would be happy to accept a PR

**Comment by tresf** [comment]
> @tresf I would be happy to accept a PR

Apparently, `inotifytools_watch_recursively_with_exclude` [assumes the exclude list is a listing of valid directories](http://inotify-tools.sourceforge.net/api/inotifytools_8c-source.html#l01282), so the  request is not possible with the built-in tools unless the utilty first built this list dynamically at runtime based on some pattern searching, which is gross.

I'm afraid this may be a `wontfix` feature and just turn into a verbosity preference.  The tool doesn't treat this folder incorrectly and there are hundreds (thousands?) of other commonly installed `/opt` locations that too would recurse into this location.

**Comment by TheAssassin** [comment]
@tresf one can always create a separate file for these kinds of features. You are hereby invited to send in a patch as proposal, we can then discuss the details.

**Comment by tresf** [comment]
I'm not a stakeholder in this feature, it was simply a cosmetic observation while debugging #430.  I won't be working on this.

**Comment by TheAssassin** [comment]
All right, thanks for the suggestion then, I'll leave this open as a feature proposal.

**Comment by azubieta** [comment]
Please transfer to the appimaged repo.
